From 80947bea470f05329af97cb792909e08ddb8843b Mon Sep 17 00:00:00 2001
From: Saurabh Sengar <saurabh.singh@xilinx.com>
Date: Mon, 11 Jun 2018 17:50:22 +0530
Subject: [PATCH 0367/1852] drm: xlnx: vtc: removing hard coded values for
 signal polarity

commit 94c68a684a6666ce1743e727cb5db41ff5890aa3 from
https://github.com/Xilinx/linux-xlnx.git

removing hard coded values for following signal polarity parameters,
and taking from vm.flags instead.
This will help clients driver to program signal polarity as per
drivers need.

	- hsync
	- vsync
	- hblank
	- vblank
	- field_id

Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_vtc.c | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_vtc.c b/drivers/gpu/drm/xlnx/xlnx_vtc.c
index 6e0b5fece3c7..72866468d671 100644
--- a/drivers/gpu/drm/xlnx/xlnx_vtc.c
+++ b/drivers/gpu/drm/xlnx/xlnx_vtc.c
@@ -272,9 +272,7 @@ struct xilinx_vtc_src_config {
 static void xilinx_vtc_config_polarity(struct xilinx_vtc *vtc,
 				       struct xilinx_vtc_polarity *polarity)
 {
-	u32 reg;
-
-	reg = xilinx_drm_readl(vtc->base, VTC_GPOL);
+	u32 reg = 0;
 
 	if (polarity->active_chroma)
 		reg |= VTC_CTL_ACP;
@@ -475,13 +473,13 @@ void xlnx_vtc_config_sig(struct xilinx_vtc *vtc,
 	xilinx_vtc_config_hori_offset(vtc, &hori_offset);
 	/* set up polarity */
 	memset(&polarity, 0x0, sizeof(polarity));
-	polarity.hsync = 1;
-	polarity.vsync = 1;
-	polarity.hblank = 1;
-	polarity.vblank = 1;
+	polarity.hsync = !!(vm->flags & DISPLAY_FLAGS_HSYNC_LOW);
+	polarity.vsync = !!(vm->flags & DISPLAY_FLAGS_VSYNC_LOW);
+	polarity.hblank = !!(vm->flags & DISPLAY_FLAGS_HSYNC_LOW);
+	polarity.vblank = !!(vm->flags & DISPLAY_FLAGS_VSYNC_LOW);
 	polarity.active_video = 1;
 	polarity.active_chroma = 1;
-	polarity.field_id = 1;
+	polarity.field_id = !!(vm->flags & DISPLAY_FLAGS_INTERLACED);
 	xilinx_vtc_config_polarity(vtc, &polarity);
 
 	/* set up src config */
-- 
2.31.1

